packages/algoliasearch-helper/src/algoliasearch.helper.js

```diff
@@ -1750,7 +1743,20 @@ AlgoliaSearchHelper.prototype._dispatchRecommendResponse = function (
   var results = {};
   content.results.forEach(function (result, index) {
     var id = ids[index];
-    results[id] = result;
+
+    if (!results[id]) {
+      results[id] = result;
+      return;
+    }
+
+    results[id].hits = results[id].hits
+      .concat(result.hits)
+      .sort(function (a, b) {
+        return b._score - a._score;
+      })
+      .filter(function (item, idx, self) {
+        return self.findIndex((i) => i.objectID === item.objectID) === idx;
+      });
```

Comment:
There are two other ways we could do this :
- Do it at the end for every id
- Calculate which ids are duplicated in `ids` and only run this for those ids.

Might be premature optimization though, this way the code is clearest.